backend/core/ai/sql_executor.py

```
@@ -0,0 +1,158 @@
+            {"valid": True/False, "error": "..."}
+        """
+        try:
+            conn = psycopg2.connect(**self.connection_params)
+            cursor = conn.cursor()
+            
+            # EXPLAIN으로 쿼리 검증 (실행은 안 함)
+            cursor.execute(f"EXPLAIN {sql}")
+            cursor.close()
+            conn.close()
+            
+            return {"valid": True}
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)